Source/WebKit/UIProcess/WebPageProxy.h

```diff
@@ -2917,10 +2918,35 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
     bool m_isListeningForUserFacingStateChangeNotification { false };
 #endif
     bool m_allowsMediaDocumentInlinePlayback { false };
-    std::unique_ptr<ProcessThrottlerActivity> m_isVisibleActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_isAudibleActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_isCapturingActivity;
-    std::unique_ptr<ProcessThrottlerActivity> m_openingAppLinkActivity;
+
+    struct ProcessActivityState {
```

Comment:
Should this go in WebPageProxyInternals.h? I wasn't sure.

Reply:
The data members used to be here so I think it is fine to have the new struct/class in this header. However, I wouldn't declare the struct/class in the middle of data members. Please move the declaration before all data members.